cmd/kube-controller-manager/app/core.go

```diff
@@ -349,20 +347,17 @@ func startGarbageCollectorController(ctx ControllerContext) (bool, error) {
 	}
 
 	gcClientset := ctx.ClientBuilder.ClientOrDie("generic-garbage-collector")
-
-	// Use a discovery client capable of being refreshed.
 	discoveryClient := cacheddiscovery.NewMemCacheClient(gcClientset.Discovery())
-	restMapper := discovery.NewDeferredDiscoveryRESTMapper(discoveryClient, meta.InterfacesForUnstructured)
-	restMapper.Reset()
+	discoveryClient.Invalidate()
```

Comment:
don't you have to put this call into the sync of the garbage collector controller?

Reply:
redundant, reset will call it